drivers/flash/nrf_qspi_nor.c

```diff
@@ -357,14 +357,13 @@ static void qspi_handler(nrfx_qspi_evt_t event, void *p_context)
 #if NRF52_ERRATA_122_PRESENT
 static bool qspi_initialized;
 
-static int anomaly_122_init(const struct device *dev)
+static nrfx_err_t anomaly_122_init(const struct device *dev)
```

Comment:
I see no benefit in changing the type of `anomaly_122_init` return value. This function is not a part of nrfx, so there is no need for it to return an `nrfx_err_t` value. 
Fixing of #39923, as mentioned in its description, requires only correction in this line: https://github.com/zephyrproject-rtos/zephyr/blob/54516743732a081deda5b32339f850bf89e69846/drivers/flash/nrf_qspi_nor.c#L772 so I don't see the point in making that many modifications as this PR does.